### PR TITLE
Switch build_tpu_quic_client to actually connect to the given validator

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Current `fn get_validator_client(&self, pubkey: &Pubkey) -> Option<QuicTpuClient>;` in `local-cluster/src/cluster.rs` always builds a client connected to the entrypoint validator, even if a different validator is specified

#### Summary of Changes
Grab the correct `ContactInfo` from the cluster and conenct to the correct valdiator 

Fixes #

### CODE REVIEW

**Review by AshwinSekar — COMMENTED** [review]
LGTM will wait for greg's approval

File: local-cluster/src/cluster.rs
Comment (gregcusack) [inline local-cluster/src/cluster.rs]: do we need these two if we can just pass in the pubkey of the entrypoint? we always know entrypoint pubkey since it is s public member of `LocalCluster`

**Review by gregcusack — COMMENTED** [review]
looks mostly look good just wondering if we need the two extra methods

File: local-cluster/src/cluster.rs
Comment (carllin) [reply local-cluster/src/cluster.rs]: haha yeah i didn't have these two at first, but a bunch of call sites were using the original default method, so figured might as well expose a convenient way to do this. 

Removed the two unnecessary methods

**Review by gregcusack — APPROVED** [review]
lgtm! thank you!

**Pull request merged**
